web/packages/teleterm/src/ui/utils/retryWithRelogin.ts

```diff
@@ -75,7 +81,13 @@ export async function retryWithRelogin<T>(
 
   const rootClusterUri = routing.ensureRootClusterUri(resourceUri);
 
-  await login(appContext, rootClusterUri);
+  if (!pendingLogin) {
+    pendingLogin = login(appContext, rootClusterUri).finally(() => {
+      pendingLogin = undefined;
+    });
+  }
+
+  await pendingLogin;
```

Comment:
I feel like this only works because we know that `pendingLogin` never rejects. Otherwise if there was a pending login, the second call would need to wait for `pendingLogin` to finish but it should ignore any errors.

But as I said, `pendingLogin` never rejects anyway. It's just something that raised my eyebrows. Maybe worth a comment.

Reply:
Maybe I should call it `pendingLoginDialog`? It would suggest that the promise is tied to a UI element (instead of an API call) so it won't reject.

Reply:
Good idea. 👍